packages/flutter/lib/src/material/mergeable_material.dart

```diff
@@ -611,7 +619,6 @@ class _MergeableMaterialState extends State<MergeableMaterial> with TickerProvid
       widgets.add(
         Container(
           decoration: BoxDecoration(
-            color: Theme.of(context).cardColor,
```

Comment:
The Google testing came back failing due to the borderRadius logic not being accounted for where the ColoredBox is now created. Since the `Container` here contains a `BoxDecoration` with a borderRadius calculated, the fill color is shaped with the mentioned corners. You will need to apply the same type of border to your change as well.  It seems to apply it to any `MaterialSlice` that comes after or before a `MaterialGap`. Here is a screenshot with the perceived differences before and after this PR. You can use your `MergeableMaterial` test on a sample app (it's easier to see in dark mode) to see this difference as well:

Before:
![Screen Shot 2020-08-24 at 5 54 56 PM](https://user-images.githubusercontent.com/27032613/91031662-7b070680-e633-11ea-8723-27ec7ef6f524.png)

After:
![Screen Shot 2020-08-24 at 5 55 10 PM](https://user-images.githubusercontent.com/27032613/91031658-79d5d980-e633-11ea-94b3-d0248ea840ac.png)

You will need to update the logic so that the border radius is applied in the way the `_borderRadius` function applies to a collection of slices, which might be tricky since this PR applies fill color to individual slices. We might also want to add a test to track that the border radius is applied to the fill color as well. It could also help for your testing purposes to [increase the border radius here](https://github.com/flutter/flutter/blob/master/packages/flutter/lib/src/material/material.dart#L61) while experimenting around with the code so you can compare the animations for what it's supposed to look like before and after your changes.

Reply:
Thanks, I'll look into it, and update the logic.

Reply:
I had a look at the code, and the only way I see is to update the logic of `_borderRadius` and how slices and gaps are added in the `widgets` list which is used as `ListBody.children`. If I'll try to improve the logic for my fix so that the `BorderRdius` is applied in the way the `_borderRadius` function applies, I'll end up adding redundancy in the code. I'll keep you updated, and ask if any help needed.

Reply:
Sounds good, thanks for the heads up!

Reply:
I did get a chance to give a closer look and play a little. I tried to apply the border-radius while we are trying are adding material slice in the [slices list](https://github.com/flutter/flutter/blob/bbfbf1770cca2da7c82e887e4e4af910034800b6/packages/flutter/lib/src/material/mergeable_material.dart#L601). 
if ( i == _children.length ) -> endRadius should be true // [this depicts ](https://github.com/flutter/flutter/blob/bbfbf1770cca2da7c82e887e4e4af910034800b6/packages/flutter/lib/src/material/mergeable_material.dart#L615)
else -> endRadius shoud be false // [this depcits if ( i < length - 2 && _children[i+1] is MaterialGap )](https://github.com/flutter/flutter/blob/bbfbf1770cca2da7c82e887e4e4af910034800b6/packages/flutter/lib/src/material/mergeable_material.dart#L543) and also if  `_children[i+1] is MaterialSlice`. For `_children[i+1] is MaterialSlice`, I'm also passing false, because the `_borderRadius` handles everything and gives us a Radius.zero. 

For the start radius, I'm checking `i==0` to verify whether the slice is the first or not which depicts the [widget.isEmpty](https://github.com/flutter/flutter/blob/bbfbf1770cca2da7c82e887e4e4af910034800b6/packages/flutter/lib/src/material/mergeable_material.dart#L543).

The result after my change
<img src="https://user-images.githubusercontent.com/34301187/92220650-eb2a4d80-eeb9-11ea-9a2d-7d5a8ceaae5c.png" width=200>

Result before my change
<img src="https://user-images.githubusercontent.com/34301187/92220758-13b24780-eeba-11ea-859b-c42869e70899.png" width=200>

There was one more issue I was facing if I was trying to refactor the code more. I was not able to figure out why it was happening, I'll write it down to you offline, and see if you can help me. I haven't run any tests apart from the visual test. I'm taking a short vacation. I'll look to run a few test and push the changes after I'm back. I thought I should inform you that I have made some progress, and it's nearly there.

Reply:
I finally got back to this, sorry for the massive delay. I don't see a difference in the two screenshots you shared in the comment. Could you elaborate? I might just be missing it because it's been so long since the last time I saw this PR

Reply:
Also, did the code with the _borderRadius logic get lost in the NNBD migrations? I don't see any of it on this PR itself

Reply:
No issues, I can totally understand. 

> I don't see a difference in the two screenshots you shared in the comment. Could you elaborate?
I posted screenshots to show that now we are able to achieve the rounded corners when color is applied, which was the issue because of which Golden Testing was failing.

>Also, did the code with the _borderRadius logic get lost in the NNBD migrations?
For a moment I got scared :/
Fortunately I haven't touched the _borderRadius logic, I have modified the logic how _borderRadius is applied. For reference you can [check this line](https://github.com/flutter/flutter/pull/63466/files#diff-9a9e318eb3a0796d3e04c28db63d9e28f8e019f3b728751d75cf3a26f2d74582R599)

Reply:
Ahh, sorry about that. I misunderstood your original comment and assumed you had made modifications to the `_borderRadius` logic itself.

Don't both screenshots depict the rounded corners? I would expect to see one screenshot with and the other without.

Reply:
Sorry but I don't remember the source of first screen shot, since it's been a long time. As far I remember, I intentionally gave border Radius in the previous implementation to get the rounded corners with color. The second screen shot is after the modification in logic of how `_borderRadius` is applied.

The both screenshot were meant to show that my modification of how `_borderRadius` is applied is same as the previous implementation. I hope it made sense, but happy to send you screenshots or add tests to clear doubts.

Reply:
That's okay! I was just a little confused because the before/after images looked identical to me and I was pretty sure I was missing something obvious and wanted to ask :)